### ISSUE TITLE
SQL Erase not working after delete

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#2607]
```clj
  (xt/submit-tx node [[:sql "INSERT INTO foo (xt$id, bar) VALUES (1, 1)"]])
  (xt/submit-tx node [[:sql "DELETE FROM foo WHERE foo.xt$id = 1"]])
  (xt/submit-tx node [[:sql "ERASE FROM foo WHERE foo.xt$id = 1"]])
  (xt/q node "SELECT * FROM foo FOR ALL VALID_TIME")
  ;; => [{:bar 1, :xt$id 1}]
```
Only looked briefly into this. There probably needs to be some more special handling in the sql-indexer for this case.

**Comment by FiV0** [comment]
Mostly fixed in c2d5e68e6ec6242aaa6021fb67d953253d491b16. Only case that remains are zero width valid-time ranges.

**Comment by jarohen** [comment]
see #2257
